services/venice-controller/src/main/java/com/linkedin/venice/controller/UnusedValueSchemaCleanupService.java

```diff
@@ -0,0 +1,129 @@
+      VeniceHelixAdmin admin,
+      VeniceParentHelixAdmin parentHelixAdmin) {
+    this.multiClusterConfig = multiClusterConfig;
+    this.admin = admin;
+    this.scheduleIntervalMinutes = multiClusterConfig.getUnusedSchemaCleanupIntervalMinutes();
+    this.minSchemaCountToKeep = multiClusterConfig.getMinSchemaCountToKeep();
+    this.veniceParentHelixAdmin = parentHelixAdmin;
+  }
+
+  private Runnable getRunnableForSchemaCleanup() {
+    return () -> {
+      if (stop) {
+        return;
+      }
+      for (String clusterName: multiClusterConfig.getClusters()) {
+        boolean cleanupEnabled =
+            multiClusterConfig.getControllerConfig(clusterName).isUnusedValueSchemaCleanupServiceEnabled();
+        if (!cleanupEnabled) {
+          continue;
+        }
+        // Get all stores for current cluster
+        List<Store> stores = admin.getAllStores(clusterName);
+        for (Store store: stores) {
+          String storeName = store.getName();
+          // Remove schema only for batch stores
+          if (isHybridStore(store) || VeniceSystemStoreUtils.isSystemStore(storeName)) {
+            continue;
+          }
+          Collection<SchemaEntry> allSchemas = veniceParentHelixAdmin.getValueSchemas(clusterName, storeName);
+
+          if (allSchemas.size() < minSchemaCountToKeep) {
+            continue;
+          }
+
+          Set<Integer> usedSchemaSet = veniceParentHelixAdmin.getInUseValueSchemaIds(clusterName, storeName);
+
+          // if any of the child colo is unreachable, skip deletion.
+          if (usedSchemaSet.isEmpty()) {
+            continue;
+          }
+          int minSchemaIdInUse = Collections.min(usedSchemaSet);
+          Set<Integer> schemasToDelete = new HashSet<>();
+
+          // assumes `getValueSchemas` returns ascending schema ids so that the older schemas are deleted first
+          for (SchemaEntry schemaEntry: allSchemas) {
+            int schemaId = schemaEntry.getId();
+            // skip latest value schema or super-set schema id
+            if (schemaId == store.getLatestSuperSetValueSchemaId() || admin.getHelixVeniceClusterResources(clusterName)
+                .getSchemaRepository()
+                .getSupersetOrLatestValueSchema(storeName)
+                .getId() == schemaId) {
+              continue;
+            }
+
+            // delete only if its not used and less than minimum of used schema id
+            if (!usedSchemaSet.contains(schemaId) && schemaId < minSchemaIdInUse) {
+              schemasToDelete.add(schemaId);
+              // maintain minimum of SCHEMA_COUNT_THRESHOLD schemas in repo
+              if (schemasToDelete.size() > allSchemas.size() - minSchemaCountToKeep) {
+                break;
+              }
+            }
+          }
+          // delete from child colos before parent
+          if (veniceParentHelixAdmin.deleteValueSchemas(clusterName, store.getName(), schemasToDelete)) {
```

Comment:
I could use admin channel, but Xun raised a point where if a child colo is unreachable, then we will delete the schema from parent even if the child colo could not delete them. I guess its fine since the admin channel will retry and delete it eventually?

Reply:
Yeah, the eventual consistency should be fine. 
For almost all the write operations, they should go through admin channel to achieve the similar guarantee about ordering and eventual consistency, otherwise, it will become hard to reason in the future.